packages/@react-aria/menu/src/useMenu.ts

```diff
@@ -68,15 +76,78 @@ export function useMenu<T>(props: AriaMenuOptions<T>, state: TreeState<T>, ref:
+  // but those felt kinda iffy. A new package?
+  // TODO: port all other translations/remove stuff if not needed
+
+  // VoiceOver has issues with announcing aria-activedescendant properly on change
+  // (especially on iOS). We use a live region announcer to announce focus changes
+  // manually. In addition, section titles are announced when navigating into a new section.
+  let stringFormatter = useLocalizedStringFormatter(intlMessages, '@react-aria/menu');
+  let focusedItem = state.selectionManager.focusedKey != null
+    ? state.collection.getItem(state.selectionManager.focusedKey)
+    : undefined;
+  let sectionKey = focusedItem?.parentKey ?? null;
+  let itemKey = state.selectionManager.focusedKey ?? null;
+  let lastSection = useRef(sectionKey);
+  let lastItem = useRef(itemKey);
+  useEffect(() => {
+    if (isAppleDevice() && focusedItem != null && itemKey !== lastItem.current) {
+      let isSelected = state.selectionManager.isSelected(itemKey);
+      let section = sectionKey != null ? state.collection.getItem(sectionKey) : null;
+      let sectionTitle = section?.['aria-label'] || (typeof section?.rendered === 'string' ? section.rendered : '') || '';
+      let announcement = stringFormatter.format('focusAnnouncement', {
+        isGroupChange: !!section && sectionKey !== lastSection.current,
+        groupTitle: sectionTitle,
+        groupCount: section ? [...getChildNodes(section, state.collection)].length : 0,
+        optionText: focusedItem['aria-label'] || focusedItem.textValue || '',
+        isSelected
+      });
+
+      announce(announcement);
+    }
+
+    lastSection.current = sectionKey;
+    lastItem.current = itemKey;
+  });
+
+  // Announce the number of available suggestions when it changes
+  let optionCount = getItemCount(state.collection);
+  let lastSize = useRef(optionCount);
+  let [announced, setAnnounced] = useState(false);
+
+  // TODO: test this behavior below, now that there isn't a open state this should just announce for the first render in which the field is focused?
+  useEffect(() => {
+    // Only announce the number of options available when the autocomplete first renders if there is no
+    // focused item, otherwise screen readers will typically read e.g. "1 of 6".
+    // The exception is VoiceOver since this isn't included in the message above.
+    let didRenderWithoutFocusedItem = !announced && (state.selectionManager.focusedKey == null || isAppleDevice());
+    if (didRenderWithoutFocusedItem || optionCount !== lastSize.current) {
+      let announcement = stringFormatter.format('countAnnouncement', {optionCount});
+      announce(announcement);
+      setAnnounced(true);
+    }
+
+    lastSize.current = optionCount;
+  }, [announced, setAnnounced, optionCount, stringFormatter, state.selectionManager.focusedKey]);
+
+  // TODO: Omitted the custom announcement for selection because we expect to only trigger onActions for Autocomplete, selected key isn't a thing
```

Comment:
For now we've decided to only handle the case where pressing on a menu item triggers an action but technically you could have a multi selection menu wrapped in the Autocomplete with which this announcement would need to be brought back